pinot-core/src/main/java/org/apache/pinot/core/operator/transform/function/ValueInTransformFunction.java

```diff
@@ -118,10 +119,9 @@ public Dictionary getDictionary() {
           _dictIdSet.add(dictId);
         }
       }
-      _dictIds = new int[DocIdSetPlanNode.MAX_DOC_PER_CALL][];
+      _dictIds = new int[length][];
```

Comment:
We should have a separate check if this in case the buffer is not big enough. Same for other methods